pkg/frontend/sku_validation.go

```diff
@@ -45,20 +45,46 @@ func validateVMSku(ctx context.Context, oc *api.OpenShiftCluster, resourceSkusCl
 	}
 
 	filteredSkus := computeskus.FilterVMSizes(skus, location)
-	masterProfileSku := string(oc.Properties.MasterProfile.VMSize)
 
-	err = checkSKUAvailability(filteredSkus, location, "properties.masterProfile.VMSize", masterProfileSku)
-	if err != nil {
+	// If the default machine size (Standard_D8s_v5) isn't available, use Standard_D8s_v4, or Standard_D8s_v3
+	err = checkSKUAvailability(filteredSkus, location, "properties.masterProfile.VMSize", string(oc.Properties.MasterProfile.VMSize))
+	if err != nil && oc.Properties.MasterProfile.VMSize == api.VMSizeStandardD8sV5 {
+		oc.Properties.MasterProfile.VMSize = api.VMSizeStandardD8sV4
+
+		err = checkSKUAvailability(filteredSkus, location, "properties.masterProfile.VMSize", string(oc.Properties.MasterProfile.VMSize))
+		if err != nil && oc.Properties.MasterProfile.VMSize == api.VMSizeStandardD8sV4 {
```

Comment:
I don't yet have the full idea, but with hard coded conditionals like these we will have more work when we have new versions / removing older versions. I was thinking instead, we could have a list of versions to iterate through for masters/workers, and have the list be ordered such we try v4 before v5. If we can do that then maybe we can also reduce the seemingly duplicated blocks of conditions.